src/commands/init.ts

```diff
@@ -0,0 +1,258 @@
+/* eslint-disable no-empty */
+import { flags } from '@oclif/command';
+import chalk from 'chalk';
+import { classToPlain } from 'class-transformer';
+import fs from 'fs';
+import inquirer from 'inquirer';
+import yaml from 'js-yaml';
+import path from 'path';
+import untildify from 'untildify';
+import Command from '../base-command';
+import DockerComposeTemplate from '../common/docker-compose/template';
+import { AccountUtils } from '../common/utils/account';
+import { ComponentConfigV1 } from '../dependency-manager/src/component-config/v1';
+import { BuildSpecV1, InterfaceSpecV1, ServiceConfigV1, ServiceVolumeV1 } from '../dependency-manager/src/service-config/v1';
+
+export abstract class InitCommand extends Command {
+  auth_required() {
+    return false;
+  }
+
+  static description = 'Initialize an architect component from an existing docker-compose file';
+
+  static flags = {
+    ...Command.flags,
+    component_file: flags.string({
+      char: 'o',
+      description: 'Path where the component file should be written to',
+      default: 'architect.yml',
+    }),
+    account: flags.string({
+      char: 'a',
+    }),
+    name: flags.string({
+      char: 'n',
+    }),
+    'from-compose': flags.string({
```

Comment:
I could have sworn that dashes didn't work, but great if they do. However, we should probably be consistent with dash or underscore.